File: order-service/internal/model/order.go
package model

import "time"

type Order struct {
	ID          uint      `gorm:"primaryKey"`
	OrderNo     string    `gorm:"unique;not null;index"`
	UserID      uint      `gorm:"not null;index"`
	ProductID   uint      `gorm:"not null"`
	ProductName string    `gorm:"not null"`
	Quantity    int       `gorm:"not null"`
	TotalPrice  float64   `gorm:"not null"`
	Status      string    `gorm:"not null;default:'pending'"` // pending, paid, completed, cancelled
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

File: frontend-web-app/src/types/order.ts
// 订单类型定义

export type OrderStatus = 'pending' | 'paid' | 'completed' | 'cancelled';

export interface Order {
  id: string;
  orderNo: string; // 订单号
  userId: string;
  productId: string;
  productName: string;
  quantity: number;
  totalPrice: number;
  status: OrderStatus;
  createdAt: string;
  updatedAt: string;
}

export interface OrderFormData {
  productId: string | number;
  quantity: number;
}

export interface CreateOrderDto {
  productId: string | number;
  quantity: number;
}

export interface OrderListResponse {
  orders: Order[];
  total: number;
  page: number;
  pageSize: number;
}
